models/Publication.js

```diff
@@ -188,6 +189,84 @@ class Publication extends BaseModel {
     return await Publication.query().patchAndFetchById(id, { deleted: date })
   }
 
+  static async update (newPubObj /*: any */) /*: Promise<any> */ {
+    // Create metadata
+    const metadata = {}
+    metadataProps.forEach(property => {
+      metadata[property] = newPubObj[property]
+    })
+
+    // Fetch the Publication that will be modified
+    let publication = await Publication.query().findById(urlToId(newPubObj.id))
+    if (!publication) {
+      return null
+    }
+
+    const modifications = _.pick(newPubObj, [
+      'id',
+      'name',
+      'description',
+      'datePublished',
+      'json',
+      'readingOrder',
+      'resources',
+      'links'
+    ])
+
+    if (metadata) {
+      modifications.metadata = metadata
+    }
+
+    if (modifications.readingOrder) {
+      modifcations.readingOrder = { data: modifications.readingOrder }
+    }
+    if (modifications.links) modifications.links = { data: modifications.links }
+    if (modifications.resources) {
+      modifications.resources = { data: modifications.resources }
+    }
+
+    // Assign the modifications to the publication object
+    publication = Object.assign(publication, modifications)
+
+    // Update Attributions if necessary
+    if (newPubObj.author) {
+      await Attribution.deleteAttributionOfPub(urlToId(newPubObj.id), 'author')
+
+      for (let i = 0; i < newPubObj.author.length; i++) {
+        await Attribution.createAttribution(
+          newPubObj.author[i],
+          'author',
+          publication
+        )
+      }
+    }
+
+    if (newPubObj.editor) {
+      await Attribution.deleteAttributionOfPub(urlToId(newPubObj.id), 'editor')
+
+      for (let i = 0; i < newPubObj.editor.length; i++) {
+        await Attribution.createAttribution(
+          newPubObj.editor[i],
+          'editor',
+          publication
+        )
+      }
+    }
```

Comment:
The problem is, we will have many more roles aside from author or editor. You should use the attributionTypes array. 
Something like 
attributionTypes.forEach((role) => { if (newPubObj[role] { ... }})